crates/ruff_linter/src/rules/perflint/rules/manual_list_comprehension.rs

```diff
@@ -346,46 +447,76 @@ fn convert_to_list_extend(
             )))
         }
         ComprehensionType::ListComprehension => {
-            let binding_stmt = binding
-                .statement(checker.semantic())
-                .and_then(|stmt| stmt.as_assign_stmt())
+            let binding_stmt_range = binding
+                .statement(semantic)
+                .and_then(|stmt| match stmt {
+                    ast::Stmt::AnnAssign(assign) => Some(assign.range),
+                    ast::Stmt::Assign(assign) => Some(assign.range),
+                    _ => None,
+                })
                 .ok_or(anyhow!(
                     "Binding must have a statement to convert into a list comprehension"
                 ))?;
-            let empty_list_to_replace = binding_stmt.value.as_list_expr().ok_or(anyhow!(
-                "Assignment value must be an empty list literal in order to replace with a list comprehension"
-            ))?;
-
-            let comprehension_body = format!("[{generator_str}]");
+            // If the binding has multiple statements on its line, only remove the list assignment
+            let binding_is_multiple_exprs = {
+                let binding_text = locator
+                    .slice(locator.full_lines_range(binding_stmt_range))
+                    .trim_whitespace_start();
+                let tokens = parse(binding_text, Mode::Module).map_err(|err| {
+                    anyhow!(
+                        "Failed to tokenize binding statement: {err} {}",
+                        binding_text
+                    )
+                })?;
+                tokens
+                    .tokens()
+                    .iter()
+                    .any(|token| matches!(token.kind(), TokenKind::Semi))
```

Comment:
I'm leaning towards not providing a fix in this case to avoid the extra complexity. 

Either way, we should avoid parsing here. We can use the `SimpleTokenizer` if we need tokenization and can't just use the AST